mri/parallel_mri/extract_sensitivity_maps.py

```diff
@@ -91,80 +61,90 @@ def extract_k_space_center_and_locations(data_values, samples_locations,
 
 
-def gridding_2d(points, values, img_shape, method='linear', point_min=None,
-                point_max=None):
+def gridded_inverse_fourier_transform_nd(kspace_loc,
+                                         kspace_data, grid, method):
     """
-    Interpolate non-Cartesian data into a cartesian grid
+    This function calculates the gridded Inverse fourier transform
+    from Interpolated non-Cartesian data into a cartesian grid
 
     Parameters
     ----------
-    points: np.ndarray
-        The 2D k_space locations of size [M, 2]
-    values: np.ndarray
-        An image of size [N_x, N_y]
-    img_shape: tuple
-        The final output ndarray
+    kspace_loc: np.ndarray
+        The N-D k_space locations of size [M, N]
+    kspace_data: np.ndarray
+        The k-space data corresponding to k-space_loc above
     method: {'linear', 'nearest', 'cubic'}, optional
         Method of interpolation for more details see scipy.interpolate.griddata
         documentation
-    points_min: float
-        The minimum points in the gridded matrix, if not provide take the min
-        of points
-    points_max: float
-        The maximum points in the gridded matrix, if not provide take the min
-        of points
+    grid: np.ndarray
+        The Gridded matrix for which you want to calculate k_space Smaps
     Returns
     -------
     np.ndarray
-        The gridded solution of shape [N_x, N_y]
+        The gridded inverse fourier transform of given kspace data
     """
-    if point_min is None:
-        point_min = np.min(points)
-    if point_max is None:
-        point_max = np.max(points)
-    xi = np.linspace(point_min,
-                     point_max,
-                     img_shape[0],
-                     endpoint=False)
-    yi = np.linspace(point_min,
-                     point_max,
-                     img_shape[1],
-                     endpoint=False)
-    grid_x, grid_y = np.meshgrid(xi, yi)
-    return griddata(points,
-                    values,
-                    (grid_x, grid_y),
-                    method=method,
-                    fill_value=0)
-
-
-def get_Smaps(k_space, img_shape, samples=None, mode='Gridding',
-              min_samples=None, max_samples=None, method='linear'):
+    gridded_kspace = griddata(kspace_loc,
+                              kspace_data,
+                              grid,
+                              method=method,
+                              fill_value=0)
+    return np.swapaxes(pfft.fftshift(
+        pfft.ifftn(pfft.ifftshift(gridded_kspace))), 1, 0)
+
+
+def get_Smaps(k_space, img_shape, samples, thresh,
+              min_samples, max_samples, mode='Gridding',
+              method='linear', n_cpu=1):
     """
     This method estimate the sensitivity maps information from parallel mri
     acquisition and for variable density sampling scheme where teh k-space
     center had been heavily sampled.
-
+    Refrence : Self-Calibrating Nonlinear Reconstruction Algorithms for
```

Comment:
Typo

Reply:
Refrence -> Reference?

Reply:
Yes sorry I should have mentioned the typo